### ISSUE TITLE
ecmwf/ecbuild/develop on discover error finding MPI for eckit in bundle

### ISSUE THREAD

**Issue by markjolah** [JCSDA/ecbuild#15]
Using ecmwf/ecbuild/develop on discover using `jedi/intel17-impi`, error finding MPI for eckit when building fv3-bundle.

```
-- [eckit] (1.4.0) [41eec82]
...
-- DEBUG - ecbuild_add_option(MPI): defaults to ON
-- DEBUG - ecbuild_evaluate_dynamic_condition(_MPI_condition): checking condition '' -> TRUE
-- DEBUG - ecbuild_add_option(MPI): ENABLE_MPI=ON was found in cache
-- DEBUG - ecbuild_add_option(MPI): defining option ENABLE_MPI 'Use system MPI libraries' ON
-- DEBUG - ecbuild_add_option(MPI): ENABLE_MPI=ON
-- DEBUG - ecbuild_add_option(MPI): feature enabled
-- DEBUG - ecbuild_add_option(MPI): searching for dependent package MPI COMPONENTS C
-- DEBUG - ecbuild_compat_require(MPI): searching for MPI - ecbuild_find_mpi(COMPONENTS C)
-- Could NOT find MPI_C (missing: MPI_C_WORKS) 
-- Could NOT find MPI (missing: MPI_C_FOUND C) 
-- Could NOT find package MPI required for feature MPI -- Provide MPI location with -DMPI_PATH=/...
CMake Error at /gpfsm/dnb32/molah/github/ecbuild/cmake/ecbuild_log.cmake:196 (message):
  CRITICAL - Feature MPI cannot be enabled -- following required
  packages weren't found: MPI
Call Stack (most recent call first):
  /gpfsm/dnb32/molah/github/ecbuild/cmake/ecbuild_add_option.cmake:239 (ecbuild_critical)
  eckit/CMakeLists.txt:30 (ecbuild_add_option)

```

Interestingly, in the same environment it seems possible to build eckit directly (not through bundle) without error.  I suspect it is something to do with languages.  Will investigate.  Possibly also a direct CMake issue.

@mmiesch Do you see similar behaviour.  (NOTE you will need bugfix/old-linker branch of ecbuild).  I will continue to investigate this afternoon.

**Comment by mmiesch** [comment]
I saw something similar with Chris H's shallow-water code that was solved by upgrading ecbuild to 3.1.0 from ecmwf.   I would have thought that version of ecbuild you're using is more recent though.

**Comment by markjolah** [comment]
Closing as it does not seem to be a current problem with JCSDA/ecbuild/develop.  Reopen if this issue persists.
